zxlive/mainwindow.py

```diff
@@ -223,6 +223,12 @@ def active_panel(self) -> Optional[BasePanel]:
 
 
     def closeEvent(self, e: QCloseEvent) -> None:
+        #Call shutdown on all tabs before closing
```

Comment:
I think you may be overthinking this. I think all that is needed to fix the issue is to add a single line in `close_tab` (just above line 329 `self.tab_widget.removeTab(i)`), like:
`widget.graph_scene.clearSelection()`

Please try to see if that works.